app/src/main/java/com/android/sample/ui/fridge/FridgeScreen.kt

```diff
@@ -30,29 +117,414 @@ import com.android.sample.ui.utils.PlateSwipeScaffold
+            shape = RoundedCornerShape(CARD_BORDER_ROUND.dp),
+            elevation = CardDefaults.cardElevation(defaultElevation = CARD_ELEVATION.dp)) {
+              Box(modifier = Modifier.fillMaxSize()) {
+                // Pencil Icon - Positioned at Top Left
+                IconButton(
+                    onClick = { showEditDialog = true },
+                    modifier =
+                        Modifier.align(Alignment.TopEnd)
+                            .padding(PADDING_8.dp)
+                            .size(EDIT_ICON_SIZE.dp),
+                ) {
+                  Icon(
+                      imageVector = Icons.Default.Edit,
+                      contentDescription = "Edit ${card.second.name} Quantity",
+                      tint = MaterialTheme.colorScheme.onSecondary,
+                  )
+                }
+
+                Column(
+                    modifier = Modifier.fillMaxSize(),
+                    horizontalAlignment = Alignment.CenterHorizontally) {
+                      Spacer(Modifier.size(EDIT_ICON_SIZE.dp).padding(PADDING_8.dp))
+
+                      Image(
+                          painter =
+                              rememberAsyncImagePainter(
+                                  model = card.second.images[PRODUCT_FRONT_IMAGE_THUMBNAIL_URL]),
+                          contentDescription = stringResource(R.string.recipe_image),
+                          modifier =
+                              Modifier.size(INGREDIENT_IMAGE_SIZE.dp)
+                                  .padding(PADDING_8.dp)
+                                  .testTag(RECIPE_IMAGE_1)
+                                  .clip(
+                                      RoundedCornerShape(
+                                          INGREDIENT_DISPLAY_IMAGE_BORDER_RADIUS.dp)),
+                          contentScale = ContentScale.Fit)
+
+                      Row(
+                          modifier =
+                              Modifier.padding(
+                                      PADDING_8.dp, PADDING_8.dp, PADDING_8.dp, PADDING_8.dp)
+                                  .background(
+                                      color = tagBackground,
+                                      shape = RoundedCornerShape(FRIDGE_TAG_CORNER.dp)),
+                          verticalAlignment = Alignment.CenterVertically,
+                          horizontalArrangement = Arrangement.Center) {
+                            Text(
+                                modifier =
+                                    Modifier.padding(
+                                        horizontal = PADDING_16.dp, vertical = PADDING_4.dp),
+                                text = "${card.first.quantity} x ${card.second.quantity}",
+                                fontSize = 14.sp,
+                                color = Color.White // Text color
+                                )
+                          }
```

Comment:
Nitpick: I feel it might be better to align with the Figma design, with the quantity at the top, the image in the middle, and the expiration bar at the bottom. What do you think?

Reply:
I initially placed it at the top, but ran into an issue where large measurements would overlap and hide the edit button. I resolved this by moving the tag to the bottom. Maybe do you have any suggestions to improve this part of the UI?

Reply:
It’s fine like that! Just make sure to update the Figma accordingly, please.